src/utils/createContextWrapper.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+/**
+ * Creates new trigger class that injects context into overlay.
+ */
+export default function createContextWrapper(Trigger, propName) {
+  return function (contextTypes) {
+    class ContextWrapper extends React.Component {
```

Comment:
A class declaration within a function is not very efficient, since this class would be redefined with every invocation. It would be better to declare this class in a separate module and instantiate it here....

Or I think I'm understanding that you are doing this because the `childContextTypes` is being appended as a static property. I'm wondering if there's at least some means to memoize this?